src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6391,3 +6396,30 @@ void os::print_memory_mappings(char* addr, size_t bytes, outputStream* st) {
     st->cr();
   }
 }
+
+bool os::read_bootid(char *dest, size_t size) {
+  int fd = ::open("/proc/sys/kernel/random/boot_id", O_RDONLY);
+  if (fd < 0) {
+    perror("CRaC: Cannot read system boot ID");
+    return false;
+  }
+  size_t rd = 0;
+  bool success = true;
+  do {
+    ssize_t r = read(fd, dest + rd, size - rd - 1);
+    if (r == 0) {
+      break;
+    } else if (r < 0) {
+      if (errno == EINTR) {
+        continue;
+      }
+      success = false;
+      perror("CRaC: Cannot read system boot ID");
+      break;
+    }
+    rd += r;
+  } while (rd < size - 1);
+  dest[size - 1] = '\0';
```

Comment:
When the function supports only `UUID_LENGTH` files it does not make much sense to zero-terminate it. It is easier to rather do `memcmp` of `UUID_LENGTH' bytes.